### PR TITLE
Stop creating new, second VCS Root. Instead use a single manually-created root in the whole project.

### PR DESCRIPTION
**Pull request by SarahFrench**
# Description

## tl;dr

This PR stops the code in `.teamcity/` creating a new [VCS Root](https://www.jetbrains.com/help/teamcity/vcs-root.html) (i.e. connection to a VCS like GitHub) and defining the build configurations in the TeamCity project to use that VCS root. Instead we'll just use the connection we make manually to establish the initial GitHub <=> TeamCity connection.

## Old workflow setting up new projects using Kotlin config code

The code we copied from folks over on `terraform-provider-azurerm` creates a VCS root in TeamCity, but to use the Kotlin code in the repo you need to create a VCS root manually to let the configuration code be used. Kind of a chicken and egg situation.

Once that connection is established via the manually created VCS root, new resources are then created by the config code that is applied. Among these resources is a second VCS root. However, the automatic syncing from the GitHub repo to the TC project (and child build configs) is still dependent on the first, manually created VCS root.

At this point you can try and neaten up the project by:
1. Disabling 'versioned settings' (i.e. syncing config changes from GitHub to TC)
2. Delete the manually made VCS root now it's not in use by the sync
3. Re-establish versioned settings using the second VCS root
4. Force a sync between GitHub and TC in response to any errors... but even this is error prone.
 
If that's confusing to read it shows how it's confusing to do 😂  I've done it successfully before, but sometimes it fails for unclear reasons. From my manual tests I think test history remains if you do a forced sync (it's not like _everything_ gets bulldozed) but secrets entered into TC are bulldozed. And generally, avoiding these risks would be nice!

## What this PR changes, and impact on workflow

This PR removes creation of the second VCS root, and instead pulls info about the manually created VCS root from TC and into a variable. This info can be accessed from TeamCity just like [how we pull in context parameter values](https://github.com/hashicorp/terraform-provider-google/blob/0b85f6252e08baf0fdb2fd5cd5c8de7147118af7/.teamcity/settings.kts#L9-L24) - using [DslContext](https://teamcity.jetbrains.com/app/dsl-documentation/root/dsl-context/index.html). This is passed into the code and allows new resources to be made so they're linked to the same VCS root.

This makes the new project setup workflow much smoother! Compared to the workflow details above, with this change setting up a new project is just:

> Once that connection is established via the manually created VCS root, new resources are then created by the config code that is applied.

Also, creating VCS roots manually makes setup of feature branch testing much, much easier. 🎉 

----
 

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [ ] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read [Write release notes](https://googlecloudplatform.github.io/magic-modules/contribute/release-notes/) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none

```

### CODE REVIEW

**Comment by SarahFrench** [comment]
https://github.com/GoogleCloudPlatform/magic-modules/assets/15078782/e6ff285b-6745-4063-a810-7cb11becff92

Here's a video showing the process of connecting a branch from the downstream repos to TeamCity using the changes in this PR.

**Review by zli82016 — APPROVED** [review]
LGTM. The changes make sense. Thanks.
